themes/bootstrap3/templates/devtools/language.phtml

```diff
@@ -9,48 +9,38 @@
 <table class="table table-striped">
   <tr><th>Language</th><th>Missing Lines</th><th>Extra Lines</th><th>Percent Translated</th><th>Extra Help Files</th></tr>
   <? foreach ($details as $langCode => $diffs): ?>
+    <?
+      $extraCount = count($diffs['notInL1']);
+      $missingCount = count($diffs['notInL2']);
+    ?>
     <tr>
       <td><?=$this->escapeHtml($langCode . ' (' . $diffs['name'] . ')')?></td>
-      <td><?=count($diffs['notInL2'])?></td>
-      <td><?=count($diffs['notInL1'])?></td>
+      <td><?=$missingCount ?><? if($missingCount > 0): ?> (<a href="#" onclick="diffManager.showMissing('<?=$langCode ?>')"><?=$this->transEsc('show') ?></a>)<? endif; ?></td>
+      <td><?=$extraCount ?><? if($extraCount > 0): ?> (<a href="#" onclick="diffManager.showExtra('<?=$langCode ?>')"><?=$this->transEsc('show') ?></a>)<? endif; ?></td>
       <td><?=$this->escapeHtml($diffs['l2Percent'])?></td>
       <td><?=count($diffs['helpFiles'])?></td>
     </tr>
   <? endforeach; ?>
 </table>
-<div class="accordion" id="accordion">
-  <? foreach ($details as $langCode => $diffs): ?>
-    <? if (count($diffs['notInL1']) > 0): ?>
-      <div class="accordion-group">
-        <div class="accordion-heading">
-          <a class="accordion-toggle" data-toggle="collapse" data-parent="#accordion" href="#extra_<?=$langCode ?>">
-            Extra Lines In <?=$this->escapeHtml($diffs['name'])?> (<?=$this->escapeHtml($langCode)?>.ini)
-          </a>
-        </div>
-        <div id="extra_<?=$langCode ?>" class="accordion-body collapse">
-          <div class="accordion-inner">
-          <? foreach ($diffs['notInL1'] as $key): ?>
-            <?=$this->escapeHtml($key)?> = "<?=$this->escapeHtml($diffs['object'][$key])?>"<br />
-          <? endforeach; ?>
-          </div>
-        </div>
-      </div>
-    <? endif; ?>
-    <? if (count($diffs['notInL2']) > 0): ?>
-      <div class="accordion-group">
-        <div class="accordion-heading">
-          <a class="accordion-toggle" data-toggle="collapse" data-parent="#accordion" href="#missing_<?=$langCode ?>">
-            Missing From <?=$this->escapeHtml($diffs['name'])?> (<?=$this->escapeHtml($langCode)?>.ini)
-            <span class="pull-right flip"></span>
-          </a>
-        </div>
-        <div id="missing_<?=$langCode ?>" class="accordion-body collapse">
-          <div class="accordion-inner">
-            <textarea class="form-control" rows="15"><? foreach ($diffs['notInL2'] as $key): ?><?=$this->escapeHtml($key)?> = "<?=$this->escapeHtml($main[$key])?>"
-<? endforeach; ?></textarea>
-          </div>
-        </div>
-      </div>
-    <? endif; ?>
-  <? endforeach; ?>
-</div>
\ No newline at end of file
+<?
+  $json = json_encode($details);
+  $script = <<<JS
+  function createDiffManager() {
+    var details = {$json};
+    function showExtra(lang) {
+      VuFind.lightbox.render('<h2>Extra Lines</h2><pre>'+details[lang].notInL1.join('\\n')+'</pre>');
```

Comment:
The lightbox approach here is definitely an improvement, except for one thing: in the old code, I used textarea tags to contain the strings, because when I submit missing strings to translators for updates, I like to click the textarea, hit ^A, ^C and paste into a text document. Using pre tags instead makes it much less convenient to select all strings for copying. Any objection to changing the tag type back? In this instance, utility is more important than attractiveness.